### ISSUE TITLE
Old Git versions cannot be installed anymore

### ISSUE THREAD

**Issue by sschuberth** [ferventcoder/chocolatey-packages#200]
Doing e.g. `cinst git -y --version 1.8.5.2` results in

    Attempt to get headers for http://msysgit.googlecode.com/files/Git-1.8.5.2-preview20131230.exe failed.
      The remote file either doesn't exist, is unauthorized, or is forbidden for url 'http://msysgit.googlecode.com/files/Git-1.8.5.2-preview20131230.exe'.

This is because Google Code has been discontinued. However, many of the old installers are still available at https://github.com/msysgit/msysgit/releases.

**Comment by ferventcoder** [comment]
@sschuberth We'll just need to go into each of the items in https://github.com/ferventcoder/chocolatey-packages/tree/master/automatic/_output/git.install and https://github.com/ferventcoder/chocolatey-packages/tree/master/automatic/_output/git.

But for some we may need to go to Chocolatey.org and use the download to fix the links (see recompile - https://chocolatey.org/docs/how-to-recompile-packages). Once fixed, they will need new version numbers - using package fix version notation - https://chocolatey.org/docs/create-packages#package-fix-version-notation

**Comment by sschuberth** [comment]
Sounds more involving that I thought. Can I ask you to work on this?

**Comment by ferventcoder** [comment]
Is it just 1.8.5.2? 

It's really not bad. I can handle it if it is just this one.

**Comment by sschuberth** [comment]
No, it's all versions < 2.0 (or more precisely, all releases listed at https://github.com/msysgit/msysgit/releases), but for now I'm mostly interested in releases "around" version 1.8.5.

**Comment by ferventcoder** [comment]
Why don't I do one and we take a look? This link looks good for me? https://github.com/msysgit/msysgit/releases/download/Git-1.8.5.2-preview20131230/Git-1.8.5.2-preview20131230.exe

**Comment by sschuberth** [comment]
Yes, that's the link to use for 1.8.5.2 now instead of the current link to http://msysgit.googlecode.com/files/Git-1.8.5.2-preview20131230.exe (also see [this](https://ci.appveyor.com/project/sschuberth/git-lfs/build/1.0.110/job/hqem6q2o1u0ase5l) failing AppVeyor CI build).

**Comment by ferventcoder** [comment]
Just a note of warning, the package version will append two numbers to the end of the fourth version segment since it is used. That will be the fixed version. Like `1.8.5.201` `01` indicating fixing this version. So folks will need to update their version choices. Just to give them a heads up.

Package versions are immutable once approved.

**Comment by sschuberth** [comment]
@ferventcoder Any update on doing the example for 1.8.5.2, so I could potentially look at the other versions?

**Comment by dragon788** [comment]
@sschuberth it looks like you could probably just copy the oldest 1.9.5 file here, https://github.com/ferventcoder/chocolatey-packages/blob/master/automatic/_output/git.install/1.9.5.20150114/tools/chocolateyInstall.ps1 and put it into a 1.8.5.2 folder with a copy of the git.install.nuspec from the same 1.9.5 version. You should be able to at least test using `choco pack` and `choco install` on a local machine.

**Comment by majkinetor** [comment]
We embedded the package and from now on older versions will always be available. This can be closed.

**Comment by sschuberth** [comment]
@majkinetor It does not seem to work (yet). https://ci.appveyor.com/project/git-lfs/git-lfs/build/1.0.943/job/q0q3rx20m6wxsa20 shows:

    git.install v1.8.5.2 [Approved]
    Attempt to get headers for http://msysgit.googlecode.com/files/Git-1.8.5.2-preview20131230.exe failed.
      The remote file either doesn't exist, is unauthorized, or is forbidden for url 'http://msysgit.googlecode.com/files/Git-1.8.5.2-preview20131230.exe'.

**Comment by majkinetor** [comment]
@sschuberth, I said, _from now on_, which means future versions wont have this problem. We wont do older packages, those will still use online installers.

The old version problem is not git specific, basically all packages suffer from it. Embedding is something we recently started to do becuase of this problem.

**Comment by sschuberth** [comment]
@majkinetor You said "from now on older versions *will always* be available", which is a misleading statement. It would be clearer if you had said "from now on, versions that will be superseded by newer versions will still be available".

In any case, then there's not point in closing this issue as it is specifically about fixing the URLs for old Git packages due to the move from Google Code to GitHub.

**Comment by majkinetor** [comment]
I am almost 100% sure that people that currently maintain Git do not have interest in providing older versions because the problem was systematic. Many people ask for the same on random tool. The best bet ist that:

- someone who badly needs it PR it.
- someobdy creates a framework that can automate this (highly improbable because vendors often change download setup between versions).

**Comment by ferventcoder** [comment]
@majkinetor I think @sschuberth would know about Git for Windows so I would lean on what is being asked for here. I'm fine with fixing older versions.

@sschuberth apologies that I haven't finished this yet. It's on my list.

**Comment by sschuberth** [comment]
@ferventcoder Sorry to both you again, but can you give an estimate on when you'll be able to address this?
